mlir/lib/Dialect/MIOpen/Transforms/AffixTuningParameters.cpp

```diff
@@ -37,6 +35,10 @@ struct AffixTuningParameters : public MIOpenOpsAffixTuningParametersPassBase<Aff
   //   This guarantess that affix tuning parameters pass generate
   //   coherent tuning parameters with the pre-set block size.
   int64_t blockSizeOverride;
+  int64_t gridSizeOverride;
+
+  // Current function
+  FuncOp func;
```

Comment:
I don't think having a pass to own the pointer to be a good idea. What if the pass is to run on multiple functions then how can we make sure this `func` member is up to date?

Reply:
My understanding is that the PassManager will call runOnFunction for each function in the module, which then walks it's operations. So if captured at the start of runOnFunction it will always be correct when applying the attributes below.

Reply:
I have no doubt about the correctness if you do `getFunction()` at the beginning. Why add the (dynamic) attribute as a class state? Using `func` as original way is cheap and saves the effort from managing the state.

Reply:
Changed to just use getFunction() since the FunctionPass already caches the current func.